lib/record_store/provider/google_cloud_dns.rb

```diff
@@ -3,7 +3,7 @@
 module RecordStore
   class Provider::GoogleCloudDNS < Provider
     class << self
-      def apply_changeset(changeset, stdout = $stdout)
+      def apply_changeset(changeset, _stdout)
```

Comment:
Complaining here from `rake test` @XaF 
```
ArgumentError: wrong number of arguments (given 1, expected 2)
    /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/provider/google_cloud_dns.rb:6:in `apply_changeset'
```

Reply:
Huh, my bad. Add a ` = nil` for the arg (as anyway you're not using it)

Reply:
Hmm, got another error after `a = nil`
Seems like it's used in NS1.rb 

15 errors have similar looking.
```
 15) Error:
NS1Test#test_txt_record_retrieved_after_adding_record_changeset:
ArgumentError: wrong number of arguments (given 2, expected 1)
    /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/provider/ns1.rb:142:in `build_from_api'
    /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/provider/ns1.rb:25:in `block in retrieve_current_records'
    /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/provider/ns1.rb:25:in `map'
    /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/provider/ns1.rb:25:in `retrieve_current_records'
    /Users/byeongyun/src/github.com/Shopify/record_store/test/providers/ns1_test.rb:430:in `block in test_txt_record_retrieved_after_adding_record_changeset'
    /Users/byeongyun/.gem/ruby/2.4.3/gems/vcr-5.0.0/lib/vcr/util/variable_args_block_caller.rb:9:in `call_block'
    /Users/byeongyun/.gem/ruby/2.4.3/gems/vcr-5.0.0/lib/vcr.rb:188:in `use_cassette'
    /Users/byeongyun/src/github.com/Shopify/record_store/test/providers/ns1_test.rb:423:in `test_txt_record_retrieved_after_adding_record_changeset'
```

Reply:
Wait, did you add `a = nil` ? I meant `= nil` after `_stdout`:

```suggestion
      def apply_changeset(changeset, _stdout = nil)
```

i.e. no need to set an useful value like `$stdout` as we're not doing anything with it anyway

Reply:
Hmm, I added it after `_stdout` like you described above. @XaF

Reply:
Are you sure this is not related to another change somewhere else ? Because that traceback above talks about `build_from_api` and not `apply_changeset`. Probably related to the removal of the `zone` argument for `build_from_api`, that was actually passed by the `retrieve_current_records` method: `full_api_records.map { |r| build_from_api(r, zone) }.flatten.compact`

Please replace `full_api_records.map { |r| build_from_api(r, zone) }.flatten.compact` by `full_api_records.map { |r| build_from_api(r) }.flatten.compact` to solve that issue.

Reply:
Thanks for your help. That works fine!